src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -11,4 +25,59 @@ public String greeting(@RequestParam(name="name", required=false, defaultValue="
 		model.addAttribute("name", name);
 		return "greeting";
 	}
+
+	@Autowired
+	private BookRepository bookRepository;
+
+	@GetMapping
+	public Iterable findAll() {
+		return bookRepository.findAll();
+	}
+
+	@GetMapping("/title/{bookTitle}")
+	public List<Book> findByTitle(@PathVariable String bookTitle) {
+		return bookRepository.findBookByTitle(bookTitle);
+	}
+
+	@GetMapping("/{id}")
+	public Book findOne(@PathVariable Long id) {
+		try {
+			return bookRepository.findBookById(id);
+		}
+		catch (BookNotFoundException ex) {
+			throw new BookNotFoundException("Book Not Found", ex);
+		}
+	}
+
+	@PostMapping
+	@ResponseStatus(HttpStatus.CREATED)
+	public Book create(@RequestBody Book book) {
+		return bookRepository.save(book);
+	}
+
+	@DeleteMapping("/{id}")
+	public void delete(@PathVariable Long id) {
+		try {
+			bookRepository.findBookById(id);
```

Comment:
Same here :) you also wouldn't need to find the book before deleting it, you can just delete it right away